### ISSUE TITLE
MzML error due to byte order mark

### ISSUE THREAD

**Issue by bernt-matthias** [compomics/searchgui#295]
I have the following error: 

```
javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,1]
Message: Content is not allowed in prolog.
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.next(XMLStreamReaderImpl.java:652)
	at com.compomics.util.experiment.io.mass_spectrometry.mzml.MzmlFileIterator.<init>(MzmlFileIterator.java:104)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileIterator.getMsFileIterator(MsFileIterator.java:70)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileHandler.writeCmsFile(MsFileHandler.java:211)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileHandler.register(MsFileHandler.java:148)
	at eu.isas.searchgui.SearchHandler$SearchWorker.doInBackground(SearchHandler.java:2536)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
java.lang.IllegalArgumentException: An exception was thrown when trying to create the mzML parser.
	at com.compomics.util.experiment.io.mass_spectrometry.mzml.MzmlFileIterator.<init>(MzmlFileIterator.java:130)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileIterator.getMsFileIterator(MsFileIterator.java:70)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileHandler.writeCmsFile(MsFileHandler.java:211)
	at com.compomics.util.experiment.io.mass_spectrometry.MsFileHandler.register(MsFileHandler.java:148)
	at eu.isas.searchgui.SearchHandler$SearchWorker.doInBackground(SearchHandler.java:2536)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

The mzML was created from raw files using OpenMS' FileConverter.

**Comment by hbarsnes** [comment]
Do you see anything special at the start of the mzML file? Are there any special characters or spaces that should not be there?  Given that it complains about the very first character I mean?

Perhaps you can also try converting the same file with another converter? For example the ThermoRawFileParser?

**Comment by bernt-matthias** [comment]
There seems to be a byte order mark in the mzml (https://en.wikipedia.org/wiki/Byte_order_mark). If I look at the file with `less` I see `<U+FEFF>`

**Comment by hbarsnes** [comment]
I assume that the byte order mark is somehow added by OpenMS then? Sounds like something that should not be there in standard mzML files?

I think we've only seen this one time before. But that was for an mgf file. Not sure whether that one also came from the OpenMS FileConverter though. And in any case, we have had no other reports of similar issues until now, hence it cannot be very common.

The byte order marker can be ignored when reading the file, but I would argue that it would be better to fix this upstream instead as I'm pretty sure we'll not be the only tool struggling with such files?

**Comment by bernt-matthias** [comment]
> Not sure whether that one also came from the OpenMS FileConverter though.

Unlikely. I tried to call trfp directly in the same cona environment. And it also produced the bom.

I guess we need to understand first why the bom is produced. I guess it's an encoding problem. If I get it right then it should not happen in utf8 https://www.opentag.com/xfaq_enc.htm

**Comment by hbarsnes** [comment]
Should be solved in ThermoRawFileParser v1.3.4 which is included in SearchGUI v4.0.28.
